synapse/__init__.py

```diff
@@ -12,8 +12,8 @@
 if (sys.version_info.major, sys.version_info.minor) < (3, 4):
     raise Exception('synapse is not supported on Python versions < 3.4')
 
-if msgpack.version < (0, 4, 2):
-    raise Exception('synapse requires msgpack >= 0.4.2')
+if msgpack.version < (0, 5, 0):
```

Comment:
`setup.py` lists `v0.5.1` as the minimum version, but `v0.5.0` is the actual lowest version that will work correctly w/ synapse.

Reply:
I would much rather fail at import time than maybe or maybe not fail sometime at runtime.

Reply:
We had a discussion in the past about updating these (and the setup.py version) and decided to not increase them at that time. This is a case where we definitely need to do it.